packages/backend/src/services/ProjectService/ProjectService.ts

```
@@ -693,19 +710,26 @@ export class ProjectService extends BaseService {
     */
     protected async getWarehouseCredentials(
         projectUuid: string,
-        userUuid: string,
+        userUuid: string | undefined,
     ) {
         let credentials =
             await this.projectModel.getWarehouseCredentialsForProject(
                 projectUuid,
             );
         let userWarehouseCredentialsUuid: string | undefined;
 
+        if (!userUuid) {
+            return {
+                ...credentials,
+                userWarehouseCredentialsUuid: undefined,
+            };
+        }
+
         if (credentials.requireUserCredentials) {
             const userWarehouseCredentials =
                 await this.userWarehouseCredentialsModel.findForProjectWithSecrets(
                     projectUuid,
-                    userUuid,
+                    userUuid ?? '',
```

Comment above relates to this. We should perhaps just throw an error instead of even trying to get user credentials when there's no user

---

@stevelikesmusic here's what I mean, if `credentials` has `requireUserCredentials` and there is no user shouldn't it fail instead of passing empty string?